### PR TITLE
Readme.md update, CSS Refactor, Moment.format(), added contributors to footer, support registrar api

### PR DESCRIPTION
**Pull request by bmiddha**
- added screenshot
- added port change info
- refactored css to a separate sass file, loaded with webpack
- added favicon

### CODE REVIEW

**Comment by bmiddha** [comment]
now uses Moment.format instead of template strings to update countdowns.
looks cleaner

**Comment by bmiddha** [comment]
added contributors to footer

**Comment by bmiddha** [comment]
added support for registrar api

**Comment by bmiddha** [comment]
last commit has introduced some problems
Building and Location, AM PM irregulartities (can be fixed with regex from scraper.ts)

File: src/getFinals.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    public courseId: Set<string>;
+    public courseTime: string;
+    public finalStart: string;
+    public finalEnd: string;
+    public finalLocations: string[];
+    public comments: string;
+}
+
+export async function getFinals(): Promise<ExamInfo[]> {
+    const examData: ExamAPI = await (await fetch("https://registrar.uic.edu/assets/scripts/finals-initial-query.php?term=220191")).json();
```
Comment (wilt00) [inline src/getFinals.ts:30]: Can we not hard code the term here?

File: frontend/main.scss [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+.countdownExpired>.finalCountdown {
+    color: #FC4A1A;    
+}
+.countdownWarning {
+    background-color: #FC4A1A;
+}
+#gradCountdown {
+    font-family: 'Roboto Mono', monospace;
+}
```
Comment (wilt00) [inline frontend/main.scss:92]: What about this requires SASS?

File: src/getFinals.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        try {
+            examInfo.finalEnd       = new Date(element.day + " 2019 " + element.time.split(" ")[2] + " " + element.time.split(" ")[3] + " CST").toISOString();
+        }
+        catch(err) {
+            console.error(err);
+            console.log(element.day + " 2019 " + element.time.split(" ")[2] + " " + element.time.split(" ")[3] + " CST");
+            console.error(examInfo);
+        }
+        return examInfo;
+    });
```
Comment (wilt00) [inline src/getFinals.ts:60]: This is not equivalent to the current behavior - doesn't deduplicate undergrad and grad finals, for example.

File: src/getFinals.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+        examInfo.courseDept         = element.course.split(" ")[0];
+        examInfo.courseNum          = element.course.split(" ")[1];
+        examInfo.courseId           = new Set().add(element.course.split(" ")[3].split("<br/>")[0]);
+        examInfo.courseTime         = element.course.split(" ")[3].split("<br/>")[1] + element.course.split(" ").slice(4).join(" ");
+        examInfo.finalLocations     = element.rooms.split(", ");
+        examInfo.comments           = element.comments;
+
+        try {
+            examInfo.finalStart     = new Date(element.day + " 2019 " + element.time.split(" ")[0] + " " + element.time.split(" ")[3] + " CST").toISOString();
```
Comment (wilt00) [inline src/getFinals.ts:44]: Year hard coded

File: src/getFinals.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            console.error(err);
+            console.log(element.day + " 2019 " + element.time.split(" ")[0] + " " + element.time.split(" ")[3] + " CST");
+            console.error(examInfo);
+        }
+        try {
+            examInfo.finalEnd       = new Date(element.day + " 2019 " + element.time.split(" ")[2] + " " + element.time.split(" ")[3] + " CST").toISOString();
+        }
+        catch(err) {
+            console.error(err);
+            console.log(element.day + " 2019 " + element.time.split(" ")[2] + " " + element.time.split(" ")[3] + " CST");
```
Comment (wilt00) [inline src/getFinals.ts:56]: This logic is repeated at least 3x - should spin out into a function

**Review by wilt00 — CHANGES_REQUESTED** [review]

**Comment by bmiddha** [comment]
Closing. No longer working on this.
New project: https://github.com/bmiddha/final-countdown

**Pull request closed without merging**